From 964740d28c9a0da52a75d579d7fdfa53df67e925 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 30 Mar 2024 16:50:46 +0100
Subject: [PATCH] src/lxc/syscall_numbers.h: drop define -1

Drop "#define -1" to avoid the following m68k build failure with gcc 12:

In file included from ../src/lxc/syscall_wrappers.h:19,
                 from ../src/lxc/mount_utils.h:15,
                 from ../src/lxc/conf.h:24,
                 from ../src/lxc/log.h:19,
                 from ../src/lxc/storage/btrfs.c:20:
../src/lxc/syscall_numbers.h:423:25: error: macro names must be identifiers
  423 |                 #define -1
      |                         ^

Fixes:
 - http://autobuild.buildroot.org/results/f40a517eba86838b11b4b1d6a8a05b8233a3394d

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Upstream: https://github.com/lxc/lxc/commit/82fe01821cd5cb8548598d7d93b07d6ef3f6b604
[Dario: make the patch to be applied with fuzz factor 0]
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
---
 src/lxc/syscall_numbers.h | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/src/lxc/syscall_numbers.h b/src/lxc/syscall_numbers.h
index ae5fdd0641c6..5dc1259fef1c 100644
--- a/src/lxc/syscall_numbers.h
+++ b/src/lxc/syscall_numbers.h
@@ -51,7 +51,6 @@
 			#define __NR_keyctl 5241
 		#endif
 	#else
-		#define -1
 		#warning "__NR_keyctl not defined for your architecture"
 	#endif
 #endif
@@ -88,7 +87,6 @@
 			#define __NR_memfd_create 5314
 		#endif
 	#else
-		#define -1
 		#warning "__NR_memfd_create not defined for your architecture"
 	#endif
 #endif
@@ -123,7 +121,6 @@
 			#define __NR_pivot_root 5151
 		#endif
 	#else
-		#define -1
 		#warning "__NR_pivot_root not defined for your architecture"
 	#endif
 #endif
@@ -158,7 +155,6 @@
 			#define __NR_setns 5303
 		#endif
 	#else
-		#define -1
 		#warning "__NR_setns not defined for your architecture"
 	#endif
 #endif
@@ -193,7 +189,6 @@
 			#define __NR_sethostname 5165
 		#endif
 	#else
-		#define -1
 		#warning "__NR_sethostname not defined for your architecture"
 	#endif
 #endif
@@ -260,7 +255,6 @@
 			#define __NR_signalfd4 5283
 		#endif
 	#else
-		#define -1
 		#warning "__NR_signalfd4 not defined for your architecture"
 	#endif
 #endif
@@ -295,7 +289,6 @@
 			#define __NR_unshare 5262
 		#endif
 	#else
-		#define -1
 		#warning "__NR_unshare not defined for your architecture"
 	#endif
 #endif
@@ -330,7 +323,6 @@
 			#define __NR_bpf 5315
 		#endif
 	#else
-		#define -1
 		#warning "__NR_bpf not defined for your architecture"
 	#endif
 #endif
@@ -365,7 +357,6 @@
 			#define __NR_faccessat 5259
 		#endif
 	#else
-		#define -1
 		#warning "__NR_faccessat not defined for your architecture"
 	#endif
 #endif
@@ -420,7 +411,6 @@
 			#define __NR_seccomp 5312
 		#endif
 	#else
-		#define -1
 		#warning "__NR_seccomp not defined for your architecture"
 	#endif
 #endif
@@ -455,7 +445,6 @@
 			#define __NR_gettid 5178
 		#endif
 	#else
-		#define -1
 		#warning "__NR_gettid not defined for your architecture"
 	#endif
 #endif
@@ -494,7 +483,6 @@
 			#define __NR_execveat 5316
 		#endif
 	#else
-		#define -1
 		#warning "__NR_execveat not defined for your architecture"
 	#endif
 #endif
@@ -733,7 +721,6 @@
 			#define __NR_personality (132 + 5000)
 		#endif
 	#else
-		#define -1
 		#warning "__NR_personality not defined for your architecture"
 	#endif
 #endif
-- 
2.43.0

